docs/source/command_line.rst

```diff
@@ -749,8 +749,14 @@ of the above sections.
 
 .. option:: --strict
 
-    This flag mode enables all optional error checking flags.  You can see the
-    list of flags enabled by strict mode in the full :option:`mypy --help` output.
+    This flag mode enables a defined subset of optional error-checking flags.
+    This subset primarily includes checks for inadvertent type unsoundness (i.e
+    strict will catch type errors as long as intentional methods like type ignore
+    or casting were not used.)
+
+    If both strict and strict-included flags are set, strict does not take precedence
+    nor override corresponding flags. You can see the list of flags enabled by strict
+    mode in the full :option:`mypy --help` output.
```

Comment:
I think `--warn-unreachable` deserves explicit mention here as something many (most?) of the users should be interested in enabling separately. It may become part of `--strict` in future, but isn't now (see #18078)